mu-plugins/rest-api/endpoints/class-wporg-themes-locale-banner-controller.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace WordPressdotorg\MU_Plugins\REST_API;
+
+use function WordPressdotorg\MU_Plugins\Helpers\Locale\{ get_all_locales_with_subdomain, get_all_valid_locales, get_locale_from_header, get_translated_locales };
+
+/**
+ * Themes_Locale_Banner_Controller
+ */
+class Themes_Locale_Banner_Controller extends Base_Locale_Banner_Controller {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wporg-themes/v1';
+		$this->rest_base = 'locale-banner';
+	}
+
+	/**
+	 * Validate the theme slug.
+	 */
+	public function check_slug( $param ) {
+		$theme = wporg_themes_theme_information( $param );
+		return ! isset( $theme->error );
+	}
+
+	/**
+	 * Get banner for general theme directory
+	 */
+	public function get_response( $request ) {
+		if ( ! defined( 'GLOTPRESS_LOCALES_PATH' ) ) {
+			return;
+		}
+
+		require_once GLOTPRESS_LOCALES_PATH;
+
+		$locale_subdomain_assoc = get_all_locales_with_subdomain();
+		$current_locale = get_locale();
+		$current_gp_locale = \GP_Locales::by_field( 'wp_locale', $current_locale );
+
+		// Build a list of WordPress locales which we'll suggest to the user.
+		$suggest_locales = array_values( array_intersect( get_locale_from_header(), get_all_valid_locales() ) );
+
+		$suggestion_links = [];
+		foreach ( $suggest_locales as $locale ) {
+			$language = \GP_Locales::by_field( 'wp_locale', $locale )->native_name;
```

Comment:
In [previous code](https://github.com/WordPress/wporg-mu-plugins/pull/609/files#diff-06764635da25e9390a050f0d6d05b3566f751bbdc8fa8f9e4afae04d17a107c5R122), we checked to make sure `$current_gp_locale` was not false which is a [possible return value](https://github.com/GlotPress/gp-locales/blob/cc5d93389d0d8ed473699900b29d31ddab27830e/locales.php#L2181). I assume we probably want to do it here as well.

Reply:
The values from `get_translated_locales` are already checked and filtered out if they return false, so we know this will be OK.